examples/life-expectancy/life-expectancy/12_ingest_table_us_gdp.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+import inspect
+import logging
+import os
+import sys
+
+import numpy as np
+import pandas as pd
+
+
+current_dir = os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe())))
+parent_dir = os.path.dirname(current_dir)
+sys.path.insert(0, parent_dir)
+
+from le_utils.utils import build_insert_query
+
+from vdk.api.job_input import IJobInput
+
+log = logging.getLogger(__name__)
+
+
+def run(job_input: IJobInput):
+    """
+    Download datasets required by the scenario and put them in the data lake.
+    """
+    log.info(f"Starting job step {__name__}")
+
+    url = "https://www.bea.gov/sites/default/files/2021-12/lagdp1221.xlsx"
+
+    df = pd.read_excel(url, header=3, na_values="(NA)").replace("'", "''", regex=True)
+
+    # select only interesting rows
+    df = df[["Unnamed: 0", 2017, 2018, 2019, 2020]]
+    df.rename(
+        {
+            "Unnamed: 0": "County",
+            2017: "Year2017",
+            2018: "Year2018",
+            2019: "Year2019",
+            2020: "Year2020",
+        },
+        axis=1,
+        inplace=True,
+    )
+    df.dropna(axis=0, inplace=True)
+
+    for i, row in df.iterrows():
+        log.info(row)
+        query = build_insert_query(row, df.columns.tolist(), "us_gdp")
```

Comment:
After I merge my latest change (PR: https://github.com/vmware/versatile-data-kit/pull/685 ) switch to 

```
job_input.send_tabular_data_for_ingestion(
      df.itertuples(index=False),
      destination_table="us_regions",
      column_names=df.columns.tolist(),
    )
```


This way Ingestion is also much much faster (due to batching of rows insertion and  parallelism)

Reply:
When I tested it I have set in the job config in 
```
db_default_type = TRINO
ingest_method_default = trino
```
 and it works 

but if you want you can also set 
```
properties_default_type=memory
```
and see if it helps. But for me it's not necessary. Can you upload all logs (from start to end) if it keeps happening.

Reply:
This is the complete log for the us_region table:

`Versatile Data Kit (VDK)
Version: 0.0.395832173
Build details: RELEASE_VERSION=0.0.395832173, BUILD_DATE=Tue Oct 26 15:15:07 UTC 2021, BUILD_MACHINE_INFO=Linux runner-mvj-ljo-project-28359933-concurrent-0v826k 5.4.129-62.227.amzn2.x86_64 #1 SMP Wed Jul 7 00:08:43 UTC 2021 x86_64 GNU/Linux, GITLAB_CI_JOB_ID=1717146199, GIT_COMMIT_SHA=cec88bc59f84e2e96ec154d288a75e09fd0dfab8, GIT_BRANCH=main

Installed plugins:
vdk-control-service-properties (from package vdk-plugin-control-cli, version 0.1.391026511)
vdk-execution-skip (from package vdk-plugin-control-cli, version 0.1.391026511)
vdk-plugin-control-cli (from package vdk-plugin-control-cli, version 0.1.391026511)
vdk-trino (from package vdk-trino, version 0.1.415648530)
vdk-ingest-file (from package vdk-ingest-file, version 0.1.377908503)
vdk-csv (from package vdk-csv, version 0.1.377908503)
vdk-server (from package vdk-server, version 0.1.379170167)
vdk-sqlite (from package vdk-sqlite, version 0.1.396520682)
vdk-ingest-http (from package vdk-ingest-http, version 0.1.377968552)
--------------------------------------------------------------------------------
Run job with directory /Users/angelica/CNR/Git/my-versatile-data-kit/examples/life-expectancy/life-expectancy
2022-02-01 17:26:45,923=1643732806[VDK] life-expectancy [DEBUG] vdk.internal.builtin_plugins.c        log_config.py:155  initialize_job  [OpId:e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805-b44b5]- Initialized logging for log type LOCAL.
2022-02-01 17:26:45,923=1643732806[VDK] life-expectancy [INFO ] vdk.plugin.control_cli_plugin. properties_plugin.py:109  initialize_job  [OpId:e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805-b44b5]- Control Service REST API URL is not configured. Will not initialize Control Service based Properties client implementation.
2022-02-01 17:26:45,923=1643732806[VDK] life-expectancy [INFO ] vdk.plugin.control_cli_plugin.    execution_skip.py:105  _skip_job_if_nec[OpId:e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805-b44b5]- Checking if job should be skipped:
2022-02-01 17:26:45,923=1643732806[VDK] life-expectancy [INFO ] vdk.plugin.control_cli_plugin.    execution_skip.py:106  _skip_job_if_nec[OpId:e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805-b44b5]- Job : life-expectancy, Team : my-team, Log config: LOCAL, execution_id: e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805
2022-02-01 17:26:45,923=1643732806[VDK] life-expectancy [INFO ] root                              execution_skip.py:111  _skip_job_if_nec[OpId:e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805-b44b5]- Local execution, skipping parallel execution check.
2022-02-01 17:26:45,923=1643732806[VDK] life-expectancy [DEBUG] vdk.internal.builtin_plugins.r          data_job.py:70   run_step        [OpId:e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805-b44b5]- Processing step 09_ingest_table_us_regions.py ...
2022-02-01 17:26:46,146=1643732806[VDK] life-expectancy [INFO ] vdk.internal.builtin_plugins.r   file_based_step.py:82   run_python_step [OpId:e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805-b44b5]- Entering 09_ingest_table_us_regions.py#run(...) ...
2022-02-01 17:26:46,146=1643732806[VDK] life-expectancy [INFO ] step_09_ingest_table_us_region 09_ingest_table_us_r:20   run             [OpId:e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805-b44b5]- Starting job step step_09_ingest_table_us_regions
2022-02-01 17:26:46,191=1643732806[VDK] life-expectancy [INFO ] vdk.internal.builtin_plugins.i   ingester_router.py:119  send_tabular_dat[OpId:e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805-b44b5]- Sending tabular data for ingestion with method: trino and target: None
2022-02-01 17:26:46,192=1643732806[VDK] life-expectancy [INFO ] vdk.internal.builtin_plugins.i     ingester_base.py:234  send_tabular_dat[OpId:e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805-b44b5]- Posting for ingestion data for table us_regions with columns ['State', 'StateCode', 'Region', 'Division'] against endpoint None
2022-02-01 17:26:46,192=1643732806[VDK] life-expectancy [DEBUG] vdk.internal.builtin_plugins.i     ingester_base.py:249  send_tabular_dat[OpId:e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805-b44b5]- Posting page 0 with 51 rows for ingestion.
2022-02-01 17:26:46,192=1643732806[VDK] life-expectancy [INFO ] vdk.internal.builtin_plugins.r   file_based_step.py:88   run_python_step [OpId:e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805-b44b5]- Exiting  09_ingest_table_us_regions.py#run(...) SUCCESS
2022-02-01 17:26:46,192=1643732806[VDK] life-expectancy [DEBUG] vdk.internal.builtin_plugins.r          data_job.py:72   run_step        [OpId:e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805-b44b5]- Processing step 09_ingest_table_us_regions.py completed successfully
2022-02-01 17:26:48,198=1643732808[VDK] life-expectancy [INFO ] vdk.plugin.trino.ingest_to_tri   ingest_to_trino.py:43   ingest_payload  [OpId:e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805-b44b5]- Ingesting payloads to table: us_regions in Trino database; collection_id: life-expectancy|e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805
2022-02-01 17:26:48,199=1643732808[VDK] life-expectancy [DEBUG] vdk.internal.builtin_plugins.c            router.py:76   open_connection [OpId:e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805-b44b5]- Connection to TRINO is missing. Will try to connect
2022-02-01 17:26:48,199=1643732808[VDK] life-expectancy [DEBUG] vdk.plugin.trino.trino_connect  trino_connection.py:56   __init__        [OpId:e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805-b44b5]- Creating new trino connection for user: root to host: localhost:8080
2022-02-01 17:26:48,200=1643732808[VDK] life-expectancy [DEBUG] vdk.plugin.trino.trino_connect  trino_connection.py:64   _connect        [OpId:e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805-b44b5]- Open Trino Connection: host: localhost:8080 with user: root; catalog: mysql; schema: life-expectancy; timeout: None
2022-02-01 17:26:48,201=1643732808[VDK] life-expectancy [DEBUG] vdk.plugin.trino.trino_connect managed_connection_b:69   connect         [OpId:e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805-b44b5]- Established <trino.dbapi.Connection object at 0x7fc728440f40>
2022-02-01 17:26:48,274=1643732808[VDK] life-expectancy [ERROR] vdk.plugin.trino.ingest_to_tri            errors.py:362  log_and_rethrow [OpId:e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805-b44b5]- A Platform service error  occurred. The error should be resolved by Platform error. Here are the details:
  WHAT HAPPENED : Failed to sent payload
WHY IT HAPPENED : Unknown error. Error message was : 'state'
   CONSEQUENCES : Will not be able to send the payload for ingestion
COUNTERMEASURES : See error message for help
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/plugin/trino/ingest_to_trino.py", line 65, in _ingest_payload
    params = tuple(obj[field] for obj in payload for field in fields)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/plugin/trino/ingest_to_trino.py", line 65, in <genexpr>
    params = tuple(obj[field] for obj in payload for field in fields)
KeyError: 'state'
2022-02-01 17:26:48,275=1643732808[VDK] life-expectancy [ERROR] vdk.internal.builtin_plugins.i     ingester_base.py:502  _payload_poster_[OpId:e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805-b44b5]- A platform error occurred while ingesting data.
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/plugin/trino/ingest_to_trino.py", line 65, in _ingest_payload
    params = tuple(obj[field] for obj in payload for field in fields)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/plugin/trino/ingest_to_trino.py", line 65, in <genexpr>
    params = tuple(obj[field] for obj in payload for field in fields)
KeyError: 'state'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/builtin_plugins/ingestion/ingester_base.py", line 478, in _payload_poster_thread
    self._ingester.ingest_payload(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/plugin/trino/ingest_to_trino.py", line 50, in ingest_payload
    self._ingest_payload(destination_table, cur, payload)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/plugin/trino/ingest_to_trino.py", line 70, in _ingest_payload
    errors.log_and_rethrow(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/core/errors.py", line 366, in log_and_rethrow
    raise to_be_raised_exception from exception
vdk.internal.core.errors.PlatformServiceError: A Platform service error  occurred. The error should be resolved by Platform error. Here are the details:
  WHAT HAPPENED : Failed to sent payload
WHY IT HAPPENED : Unknown error. Error message was : 'state'
   CONSEQUENCES : Will not be able to send the payload for ingestion
COUNTERMEASURES : See error message for help
2022-02-01 17:26:48,279=1643732808[VDK] life-expectancy [ERROR] vdk.internal.builtin_plugins.i            errors.py:280  log_and_throw   [OpId:e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805-b44b5]- A Platform service error  occurred. The error should be resolved by Platform error. Here are the details:
  WHAT HAPPENED : Failed to post all data for ingestion successfully.
WHY IT HAPPENED : Payloads failed to get posted for ingestion
   CONSEQUENCES : Some data will not be ingested.
COUNTERMEASURES : There has been temporary failure. You can retry the data job again. If error persist inspect logs and check ingest plugin documentation.
Traceback (most recent call first):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/builtin_plugins/ingestion/ingester_base.py", line 587, in _log_and_throw
    errors.log_and_throw(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/builtin_plugins/ingestion/ingester_base.py", line 574, in _handle_results
    self._log_and_throw(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/builtin_plugins/ingestion/ingester_base.py", line 548, in close_now
    self._handle_results()
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/builtin_plugins/ingestion/ingester_base.py", line 539, in close
    self.close_now()
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/builtin_plugins/ingestion/ingester_router.py", line 242, in close
    ingester.close()
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/builtin_plugins/ingestion/ingester_configuration_plugin.py", line 38, in finalize_job
    context.ingester.close()
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/builtin_plugins/run/data_job.py", line 267, in run
    self._plugin_hook.finalize_job(context=job_context)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/builtin_plugins/run/cli_run.py", line 65, in run_job
    execution_result = job.run(args)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/builtin_plugins/run/cli_run.py", line 116, in run
    run_impl.run_job(context, pathlib.Path(data_job_directory), arguments)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/cli_entry.py", line 95, in vdk_cli_execute
    exit_code = root_command(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/cli_entry.py", line 132, in vdk_main
    exit_code = cast(InternalHookSpecs, plugin_registry.hook()).vdk_cli_execute(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/cli_entry.py", line 167, in main
    exit_code = cast(InternalHookSpecs, plugin_registry.hook()).vdk_main(
  File "/Library/Frameworks/Python.framework/Versions/3.8/bin/vdk", line 8, in <module>
    sys.exit(main())

2022-02-01 17:26:48,279=1643732808[VDK] life-expectancy [WARNI] vdk.internal.builtin_plugins.i   ingester_router.py:245  close           [OpId:e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805-b44b5]- Closing of ingester for method `trino` failed. Exception was: A Platform service error  occurred. The error should be resolved by Platform error. Here are the details:
  WHAT HAPPENED : Failed to post all data for ingestion successfully.
WHY IT HAPPENED : Payloads failed to get posted for ingestion
   CONSEQUENCES : Some data will not be ingested.
COUNTERMEASURES : There has been temporary failure. You can retry the data job again. If error persist inspect logs and check ingest plugin documentation.
2022-02-01 17:26:48,279=1643732808[VDK] life-expectancy [ERROR] vdk.internal.builtin_plugins.r            errors.py:362  log_and_rethrow [OpId:e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805-b44b5]- A Platform service error  occurred. The error should be resolved by Platform error. Here are the details:
  WHAT HAPPENED : Failed executing job.
WHY IT HAPPENED : An exception occurred, exception message was: Ingesting data failed
  WHAT HAPPENED : On close some following ingest queues types reported errors:  ['trino'].
WHY IT HAPPENED : There were errors while closing ingestion. Exceptions were: {'trino': PlatformServiceError('A Platform service error  occurred. The error should be resolved by Platform error. Here are the details:\n  WHAT HAPPENED : Failed to post all data for ingestion successfully.\nWHY IT HAPPENED : Payloads failed to get posted for ingestion\n   CONSEQUENCES : Some data will not be ingested.\nCOUNTERMEASURES : There has been temporary failure. You can retry the data job again. If error persist inspect logs and check ingest plugin documentation.')}.
   CONSEQUENCES : Some data was partially ingested or not ingested at all.
COUNTERMEASURES : Follow the instructions in the error messages and log warnings. Make sure to inspect any errors or warning logs generatedRe-try the job if necessary
   CONSEQUENCES : The provided Data Job will not be executed. Terminating application.
COUNTERMEASURES : See contents of the exception and fix the problem that causes it. Most likely a prerequisite or plugin of one of the key VDK components failed, see logs for details and ensure the prerequisite for the failed component (details in stacktrace).
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/builtin_plugins/run/cli_run.py", line 65, in run_job
    execution_result = job.run(args)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/builtin_plugins/run/data_job.py", line 267, in run
    self._plugin_hook.finalize_job(context=job_context)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/builtin_plugins/ingestion/ingester_configuration_plugin.py", line 38, in finalize_job
    context.ingester.close()
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/builtin_plugins/ingestion/ingester_router.py", line 272, in close
    raise PlatformServiceError(message)
vdk.internal.core.errors.PlatformServiceError: Ingesting data failed
  WHAT HAPPENED : On close some following ingest queues types reported errors:  ['trino'].
WHY IT HAPPENED : There were errors while closing ingestion. Exceptions were: {'trino': PlatformServiceError('A Platform service error  occurred. The error should be resolved by Platform error. Here are the details:\n  WHAT HAPPENED : Failed to post all data for ingestion successfully.\nWHY IT HAPPENED : Payloads failed to get posted for ingestion\n   CONSEQUENCES : Some data will not be ingested.\nCOUNTERMEASURES : There has been temporary failure. You can retry the data job again. If error persist inspect logs and check ingest plugin documentation.')}.
   CONSEQUENCES : Some data was partially ingested or not ingested at all.
COUNTERMEASURES : Follow the instructions in the error messages and log warnings. Make sure to inspect any errors or warning logs generatedRe-try the job if necessary
2022-02-01 17:26:48,279=1643732808[VDK] life-expectancy [ERROR] vdk.internal.cli_entry                 cli_entry.py:145  vdk_main        [OpId:e4005869-7cf0-4f49-8df5-40f69f841a0f-1643732805-b44b5]- Exiting with exception.
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/cli_entry.py", line 132, in vdk_main
    exit_code = cast(InternalHookSpecs, plugin_registry.hook()).vdk_cli_execute(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/cli_entry.py", line 95, in vdk_cli_execute
    exit_code = root_command(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/core.py", line 1259, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/click/decorators.py", line 21, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/builtin_plugins/run/cli_run.py", line 116, in run
    run_impl.run_job(context, pathlib.Path(data_job_directory), arguments)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/builtin_plugins/run/cli_run.py", line 70, in run_job
    errors.log_and_rethrow(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/core/errors.py", line 368, in log_and_rethrow
    raise exception
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/builtin_plugins/run/cli_run.py", line 65, in run_job
    execution_result = job.run(args)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/builtin_plugins/run/data_job.py", line 267, in run
    self._plugin_hook.finalize_job(context=job_context)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/callers.py", line 208, in _multicall
    return outcome.get_result()
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/builtin_plugins/ingestion/ingester_configuration_plugin.py", line 38, in finalize_job
    context.ingester.close()
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/vdk/internal/builtin_plugins/ingestion/ingester_router.py", line 272, in close
    raise PlatformServiceError(message)
vdk.internal.core.errors.PlatformServiceError: Ingesting data failed
  WHAT HAPPENED : On close some following ingest queues types reported errors:  ['trino'].
WHY IT HAPPENED : There were errors while closing ingestion. Exceptions were: {'trino': PlatformServiceError('A Platform service error  occurred. The error should be resolved by Platform error. Here are the details:\n  WHAT HAPPENED : Failed to post all data for ingestion successfully.\nWHY IT HAPPENED : Payloads failed to get posted for ingestion\n   CONSEQUENCES : Some data will not be ingested.\nCOUNTERMEASURES : There has been temporary failure. You can retry the data job again. If error persist inspect logs and check ingest plugin documentation.')}.
   CONSEQUENCES : Some data was partially ingested or not ingested at all.
COUNTERMEASURES : Follow the instructions in the error messages and log warnings. Make sure to inspect any errors or warning logs generatedRe-try the job if necessary`

Reply:
Can you please upgrade vdk in your environment? 

Those logs are from older version. You have vdk-trino version [0.1.415648530](https://pypi.org/project/vdk-trino/0.1.415648530)  while latest version is [0.1.460149153](https://pypi.org/project/vdk-trino/0.1.460149153)

Use 
```
pip install --upgrade --upgrade-strategy eager quickstart-vdk  vdk-trino`

Reply:
Problem solved. Thanks.